release/air_tests/air_benchmarks/workloads/train_benchmark.py

```diff
@@ -0,0 +1,163 @@
+import ray
+from ray.air import session
+from ray.train.torch import TorchTrainer
+from ray.air.config import ScalingConfig
+
+import time
+import os
+import json
+
+import torch
+import torchvision
+
+DEFAULT_IMAGE_SIZE = 224
+
+# This benchmark does the following:
+# 1) Read images with ray.data.read_images()
+# 2) Apply preprocessing with map_batches()
+# 3) Train TorchTrainer on processed data
+# The resulting metrics are:
+# - ray.data+transform: Throughput of data ingest (steps 1+2 above)
+# - ray.torchtrainer.fit: Throughput of the final epoch in
+#   TorchTrainer.fit() (step 3 above)
+
+
+def iterate(dataset, label, metrics):
+    start = time.time()
+    it = iter(dataset)
+    num_rows = 0
+    for batch in it:
+        num_rows += len(batch)
+    end = time.time()
+    print(label, end - start, "epoch", i)
+
+    tput = num_rows / (end - start)
+    metrics[label] = tput
+
+
+def get_transform(to_torch_tensor):
+    # Note(swang): This is a different order from tf.data.
+    # torch: decode -> randCrop+resize -> randFlip
+    # tf.data: decode -> randCrop -> randFlip -> resize
+    transform = torchvision.transforms.Compose(
+        [
+            torchvision.transforms.RandomResizedCrop(
+                size=DEFAULT_IMAGE_SIZE,
+                scale=(0.05, 1.0),
+                ratio=(0.75, 1.33),
+            ),
+            torchvision.transforms.RandomHorizontalFlip(),
+        ]
+        + [torchvision.transforms.ToTensor()]
+        if to_torch_tensor
+        else []
+    )
+    return transform
+
+
+def crop_and_flip_image_batch(image_batch):
+    transform = get_transform(False)
+    batch_size, height, width, channels = image_batch["image"].shape
+    tensor_shape = (batch_size, channels, height, width)
+    image_batch["image"] = transform(
+        torch.Tensor(image_batch["image"].reshape(tensor_shape))
+    )
+    return image_batch
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--data-root",
+        default="s3://air-cuj-imagenet-1gb",
+        type=str,
+        help="Directory path with files.",
+    )
+    parser.add_argument(
+        "--batch-size",
+        default=32,
+        type=int,
+        help="Batch size to use.",
+    )
+    parser.add_argument(
+        "--num-epochs",
+        default=2,
+        type=int,
+        help="Number of epochs to run. The throughput for the last epoch will be kept.",
+    )
+    parser.add_argument(
+        "--num-workers",
+        default=2,
```

Comment:
should we test on a larger scale as well?

Reply:
I think the default value can be small, to ease local testing. But we should use a larger arg in `release_tests.yaml`.